benchmark/C/Savina/src/micro/ThreadRing.lf

```diff
@@ -152,14 +160,16 @@ reactor Initializer(numReactors:int(100), numPings:int(100000)) {
     =}
 }
 
-main reactor (numIterations:int(12), numPings:int(100000)) {
-    
-    /* [[[cog
-      cog.outl(f'init = new Initializer(numReactors={numReactors}, numPings=numPings);')
-    ]]] */
-    init = new Initializer(numReactors=100, numPings=numPings);
-    /// [[[end]]]    
+/*[[[cog
+cog.outl(f'main reactor (numPings:int({numPings}), numReactors:int({numReactors}))')
+]]] */
+main reactor (numPings:int(100000), numReactors:int(100)) 
+
+/// [[[end]]]
+
+{
     
+    init = new Initializer(numReactors=numReactors, numPings=numPings);    
     
     reaction(startup) -> init.inStart {=
         SET(init.inStart, true);
```

Comment:
This comment refers to code below (Unfortunately, I can only comment on code that changed). We have better mechanisms for connecting a large number of reactors now. Please have a look at the updated C++ version for the required pattern.

Reply:
Is there  a special syntax for this in the C target?  When I tried using the C++ pattern the compiler reports "extraneous input '-1' expecting ']" in reference to this line:

workers = new[numReactors-1] ThreadRingReactor();

Is this expected?

Reply:
AFAIK, the `new` construct does not accept arbitrary arithmetic expressions. I think it will accept sums of constants or parameters, but not arbitrary arithmetic expressions.  And even those sums are not documented in the [language specification](https://github.com/lf-lang/lingua-franca/wiki/Multiports-and-Banks-of-Reactors#banks-of-reactors).

Reply:
Our syntax doesn't allow this. The C++ version uses the target code escape `{=...=}` to do the calculation in target code:
```
workers = new[{=numReactors-1=}] ThreadRingReactor();
```
This, however, is only supported by the C++ target and does not work in C. I think your only option is to use cog here. So you would do the subtraction in python code and generate the line accordingly.